apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,514 @@
+        [InlineData("0.00000009", "0.0000008")]
+        [InlineData("1.999999999999999999999999", "2")]
+        public void CompareToLessThanTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.True(result < 0);
+        }
+
+        [Theory]
+        [InlineData("NaN", "99999999999999999999999999999")]
+        [InlineData("99.99", "99.90")]
+        [InlineData("1.0", "-1.000")]
+        [InlineData("-1", "-123.0")]
+        [InlineData("0", "-0.0009")]
+        [InlineData("0.00009", "0.00000009")]
+        public void CompareToGreaterThanTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.True(result > 0);
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123", "123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123")]
+        [InlineData("-123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123", "-123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("-0.00", "0.0")]
+        public void GetHashCodeTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var hashCode1 = number1.GetHashCode();
+            var hashCode2 = number2.GetHashCode();
+            Assert.Equal(hashCode1, hashCode2);
+        }
+
+        [Theory]
+        [InlineData("1.00")]
+        [InlineData("1.0")]
+        [InlineData("NaN")]
+        [InlineData("1.0000")]
+        [InlineData("-1.00")]
+        [InlineData("-0.00")]
+        [InlineData("0.0000000000000009")]
+        [InlineData("-0.0000000000000009")]
+        [InlineData("1234567891234567891234567891234567891234567891234567891234567891234567891234567891234567891230.12345678912345678912345678912345678912345678")]
+        [InlineData("-1234567891234567891234567891234567891234567891234567891234567891234567891234567891234567891230.12345678912345678912345678912345678912345678")]
+        public void ParseValidTest(string input)
+        {
+            var number = PgNumeric.Parse(input);
+            Assert.IsType<PgNumeric>(number);
+            Assert.NotNull(number.ToString());
+        }
+
+        [Theory]
+        [MemberData(nameof(GetStringInputs))]
+        public void ParseTest(string input, bool expectedParsingResult)
```

Comment:
Remove this one, you now have tests for parsing success and tests for parsing failures.
Remove the data member as well, if you want to keep some cases from the data member you can add them to the specific tests, maybe creating a data member for each instead of using inline data.